### PR TITLE
XW-1690 | support breakpoints in global navigation

### PR DESCRIPTION
**Pull request by tomasznapieralski**
https://wikia-inc.atlassian.net/browse/XW-1690
https://github.com/Wikia/design-system-reference-page/pull/70

@Wikia/x-wing

### CODE REVIEW

File: components/_global-navigation.scss [hunk 1]
```diff
@@ -264,24 +277,65 @@ $notification-counter-size: 18px;
...
+		padding: 8px 10px;
+	}
+
+	&__start-a-wiki-caption {
+		display: none;
+	}
+
+	&__start-a-wiki-icon {
+		display: block;
+		height: 14px;
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: Maybe we could use `square` mixin here?

File: components/_global-navigation.scss [hunk 2]
```diff
@@ -200,17 +200,26 @@ $notification-counter-size: 18px;
...
-		.wds-global-navigation__search-input-wrapper {
-			margin-right: 15px;
-		}
+		.wds-global-navigation__search {
+			flex: 2;
 
-		.wds-global-navigation__search-input::placeholder {
-			color: $color-light-gray;
-		}
+			&-input-wrapper {
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: It's really hard to be grepped now. What do you think about using this pattern only after double-underscore?

File: components/_global-navigation.scss [hunk 3]
```diff
@@ -264,24 +277,65 @@ $notification-counter-size: 18px;
...
+		&__search {
+			flex: 2;
+
+			&-input-wrapper {
+				border-bottom: 1px solid $color-slate-gray;
+			}
+
+			&-input {
+				display: block;
+			}
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: I see `-input-wrapper` and `-input` styles repeated in active state and on this breakpoint. What do you think about overriding this style only for mobile not active state?

File: components/_global-navigation.scss [hunk 4]
```diff
@@ -264,24 +277,65 @@ $notification-counter-size: 18px;
...
+
+	&__start-a-wiki-icon {
+		display: block;
+		height: 14px;
+		width: 14px;
+	}
+
+	@media #{$breakpoint-desktop-up} {
+		&__content-bar {
+			width: 1024px;
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: Please make sure those numbers are accepted by Design

File: components/_global-navigation.scss [hunk 5]
```diff
@@ -11,71 +11,81 @@ $notification-counter-size: 18px;
 	border-width: $border-top-width 0 0;
 	box-shadow: 0 1px 0 0 $color-faint-gray;
 
-	&__content-bar {
-		height: $nav-height;
-		margin: 0 auto;
-		position: relative;
-		width: 1024px - $scrollbar-width;
-		z-index: 1;
+	@media only screen and (max-width: 1023px) {
```
Comment (hakubo) [inline components/_global-navigation.scss]: 1. Can we define these breakpoints in one place?
2. Did we decide to have such breakpoint?

File: components/_global-navigation.scss [hunk 6]
```diff
@@ -263,30 +280,123 @@ $notification-counter-size: 18px;
...
 		}
 
-		&__start-a-wiki::before {
-			content: '┼';
-			display: inline;
+		&__content-item-search {
+			flex: 1;
+		}
+
+		&__content-item-search-input-wrapper {
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: can we get rid of this `content-item-` prefix?

Comment (Warkot) [inline components/_global-navigation.scss]: https://github.com/Wikia/design-system/pull/44/commits/b06cebaa5ba470a504338d6d1ff75ea58fb04eeb

File: components/_global-navigation.scss [hunk 7]
```diff
@@ -263,30 +280,123 @@ $notification-counter-size: 18px;
...
+			display: none;
+		}
+
+		&.wds-search-is-active .wds-global-navigation {
+			&__content-item-search-input-wrapper {
+				margin-right: 15px;
+			}
+
+			&__content-item-search-input::placeholder {
+				color: $color-light-gray;
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: Do we need to override this value?

File: components/_global-navigation.scss [hunk 8]
```diff
@@ -263,30 +280,123 @@ $notification-counter-size: 18px;
...
+		&__content-item-account-menu-caption,
+		&__content-item-start-a-wiki-caption {
+			display: inline;
+		}
+
+		&__content-item-start-a-wiki-icon {
+			display: none;
+		}
+
+		&.wds-search-is-active .wds-global-navigation {
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: Probably entire block is not required as it overrides the same values.

File: components/_global-navigation.scss [hunk 9]
```diff
@@ -275,30 +280,104 @@ $notification-counter-size: 18px;
 		top: 7px;
 	}
 
-	&__register-button-caption {
-		font-size: $typescale-size-minus-2;
-		padding-bottom: 10px;
-		text-align: center;
+	&__start-a-wiki .wds-button {
+		padding: 8px 10px;
```
Comment (kvas-damian) [inline components/_global-navigation.scss]: Please update button component to match new designs.

**Comment by kvas-damian** [comment]
LGTM

File: base/_breakpoints.scss [hunk 10]
```diff
@@ -16,6 +16,9 @@ $breakpoint-tablet-down: "#{$screen} and (max-width:#{$breakpoint-desktop - 1})"
 $breakpoint-tablet-only: "#{$screen} and (min-width:#{$breakpoint-tablet}) and (max-width:#{$breakpoint-desktop - 1})";
 
 $breakpoint-desktop-up: "#{$screen} and (min-width:#{$breakpoint-desktop})";
-$breakpoint-desktop-only: "#{$screen} and (min-width:#{$breakpoint-desktop}) and (max-width:#{$breakpoint-desktop-xl - 1})";
+$breakpoint-desktop-only: "#{$screen} and (min-width:#{$breakpoint-desktop}) and (max-width:#{$breakpoint-desktop-l - 1})";
+
+$breakpoint-desktop-l-up: "#{$screen} and (min-width:#{$breakpoint-desktop-l})";
```
Comment (hakubo) [inline base/_breakpoints.scss]: l?
large?

**Comment by hakubo** [comment]
one minor comment, LGTM

**Pull request merged**